src/search/hnsw_indexer.h

```diff
@@ -41,16 +41,17 @@ struct HnswNode {
 
   HnswNode(NodeKey key, uint16_t level);
 
-  StatusOr<HnswNodeFieldMetadata> DecodeMetadata(const SearchKey& search_key, engine::Storage* storage) const;
+  StatusOr<HnswNodeFieldMetadata> DecodeMetadata(engine::Context& ctx, const SearchKey& search_key,
```

Comment:
We can remove the `storage` parameter here?

Reply:
storage can be obtained through ctx

Reply:
oh, yes. What I want to say is to remove this storage in this and following methods since we can get the storage via the context.